File: grind75-05-valid-palindrome.py
# https://leetcode.com/problems/valid-palindrome
class Solution:
    # O(n) time
    def isPalindrome(self, s: str) -> bool:
        s_plain = ''.join(filter(str.isalnum, s)).lower()
        return s_plain == s_plain[::-1]

File: grind75-05-valid-palindrome.ts
// https://leetcode.com/problems/valid-palindrome/

// O(n) - 70th percentile
function isPalindrome(s: string): boolean {
    const ALPHANUM = new Set('abcdefghijklmnopqrstuvwxyz0123456789')
    // not using flat str here b/c its immutability makes looped '+=' inefficient
    let charList: string[] = []
    for (let char of s.toLowerCase()) {
        if (ALPHANUM.has(char)) {
            charList.push(char)
        }
    }
    return charList.join('') === charList.reverse().join('')
}
